### DISCUSSION TITLE
Expert/Normal settings layout

### DISCUSSION THREAD

**Discussion by fast4x** [fast4x/RiMusic#2049] [Ideas]
This discussion is related to all ideas to implement new settings optimization.

It is useful to write a list of settings to have in expert and normal mode.

Space to creativity, thanks in advance.

**Comment by NikunjK8** [comment]
All the settings existed in ViMusic by default goes to normal mode for people switching over.

**Reply by ikanakova** [reply]
I agree. Then it should stop being called ViMusic mode. Actually, it would be Default mode, and when Expert mode is enabled, the user could change the look of the player to their liking.

**Comment by dAtA-TRoN** [comment]
> This discussion is related to all ideas to implement new settings optimization.
> 
> It is useful to write a list of settings to have in expert and normal mode.
> 
> Space to creativity, thanks in advance.
> 

Unfortunately, I won't be able to assist much as I'm currently in the hospital (no, it's nothing serious 🙄😄). But I have faith in your expertise 🫡.

**Reply by fast4x** [reply]
> > This discussion is related to all ideas to implement new settings optimization.
> > 
> > It is useful to write a list of settings to have in expert and normal mode.
> > 
> > Space to creativity, thanks in advance.
> > 
> 
> Unfortunately, I won't be able to assist much as I'm currently in the hospital (no, it's nothing serious 🙄😄). But I have faith in your expertise 🫡.

Oh man, I’m sorry you’re in the hospital, I hope you work it out soon.

**Reply by dAtA-TRoN** [reply]
Thanks a bunch! I'll probably be out of here soon enough. After all, gotta make sure RiMusic turns into a diamond someday. And you know as I always say, 'fast4x is good, but with my help, he can become the best 🤣🤣🤣🤣🤣🤣🤣🤣!'

**Reply by fast4x** [reply]
Where I come from, it’s write "MINCHIA how modest you are!" 😂😂😂

MINCHIA = sicilian universal term 😂😂🤣🤣

**Comment by airi019** [comment]
1. merge UI and Player appearance into a single "Appearance"

2. move most options from "player" section in the UI group into a new group called "Player" (persistent queue, resume playback, skip silence, loudness normalization, etc)

3. data should be moved into the advanced group and the default value for song cache should be unlimited. Since this is a music app, I doubt anyone would ever wipe their cache, that is why cache and download should become the same thing and cached songs should automatically become "Downloaded".

4. "check update" should be in the about group just like all other apps and enabled by default if you installed from izzyondroid

5. skip silence and loudness normalization should be easily accessible from the player screen just like the repeat and shuffle button

**Reply by ikanakova** [reply]
In my opinion, it should first be decided what settings will be only in Expert mode and what settings will be in Default mode, and then only if necessary to override the logic of the settings...

**Reply by ikanakova** [reply]
> 4. "check update" should be in the about group just like all other apps and enabled by default if you installed from izzyondroid

And I definitely don't agree with enabled by default. This is the thing that always bothers me the most about apps...

**Reply by fast4x** [reply]
agree... by default no... in the future would be add a form to ask user...

**Reply by Sensex00** [reply]
> agree... by default no... in the future would be add a form to ask user...

Yes you can add ask to check for update button in settings

**Reply by airi019** [reply]
> In my opinion, it should first be decided what settings will be only in Expert mode and what settings will be in Default mode, and then only if necessary to override the logic of the settings...

my image of "expert" means developer and debugging options
I don't get what you mean
as far as I could see there are no settings you should hide from the user

**Reply by ikanakova** [reply]
> > In my opinion, it should first be decided what settings will be only in Expert mode and what settings will be in Default mode, and then only if necessary to override the logic of the settings...
> 
> my image of "expert" means developer and debugging options I don't get what you mean as far as I could see there are no settings you should hide from the user

I meant that there will be 2 modes of settings (like in VLC on PC) - the basic Default mode for users who don't want to set anything much and just use the app, and Expert mode where people like me and NikunjKhangwal and dAtA-TRoN can tweak the settings to their satisfaction.

**Comment by Sensex00** [comment]
Whole app port to material design 3 ui and guidelines i mean app uses many things from m3 but in bits and pieces this would make your app more consistent in many ways it will be not inconsistent

Example player ui appearance by making some customisation can see various inconsistencies and problems

**Reply by fast4x** [reply]
RiMusic uses many M3 components, but tries not to conform to all other apps. At most it will have a theme more material.

You can free explain your ideas and share it with we

**Reply by Sensex00** [reply]
I said this bcz I only want want ui to be more neat and consistence you do whatever to archive this thanks you by not going all m3

**Reply by fast4x** [reply]
Don’t worry, express your idea and we’ll talk

**Reply by Sensex00** [reply]
Hey it's a silly question by why are not at any point thought ui like vimusic but vitune does mimic same ui by default

**Reply by fast4x** [reply]
RiMusic does not imitate ViMusic, extends it

**Reply by Sensex00** [reply]
Hey don't take this disrespectfully do you think if you have imitate vimusic starting or at any point  there will be less issues centred around ui

**Reply by fast4x** [reply]
I do not find it disrespectful, in this project anyone can express their ideas in respect of the ideas of others.

**Reply by Sensex00** [reply]
Answer your thought

**Reply by ikanakova** [reply]
I think we should get rid of ViMusic completely, which is a long dead project anyway, and let fast4x keep creating RiMusic as a unique fantastic player 💚

**Reply by fast4x** [reply]
Migration is just in progress..

**Reply by NikunjK8** [reply]
Just let me have those ViMusic traits like floating search and center-aligned artist/album names without icons.
If they gets implemented as a seperate customization options then ViMusic mode can go

**Reply by fast4x** [reply]
> Just let me have those ViMusic traits like floating search and center-aligned artist/album names without icons.
> If they gets implemented as a seperate customization options then ViMusic mode can go

In the future ViMusic term will be Essential in all settings part with the actual ViMusic configuration.

**Comment by fast4x** [comment]
I'm very happy for this discussion. Thanks to all.

**Reply by Sensex00** [reply]
Hey does it uses coil https://coil-kt.github.io/coil/

**Reply by ikanakova** [reply]
> Hey does it uses coil https://coil-kt.github.io/coil/

It's not clear to me what good it would do in RiMusic...

**Reply by fast4x** [reply]
Also i don't understand, coil is used in RiMusic for load and cache images

**Reply by ikanakova** [reply]
I now understand the role of Coil in RiMusic (I'm always learning new things about RiMusic 🤩). But I still don't see how this @Sensex00 question is relevant to this discussion.

**Reply by Sensex00** [reply]
> I now understand the role of Coil in RiMusic (I'm always learning new things about RiMusic 🤩). But I still don't see how this @Sensex00 question is relevant to this discussion.

I finded it another app i thought it was good so I wanted find it being using or not so it would good alternative

**Reply by Sensex00** [reply]
Hey why when I listen to one particular song in album  besides artist name is YouTube logo not artists logo why

**Reply by fast4x** [reply]
it occur when YouTube music code isn't available...

**Reply by Sensex00** [reply]
I tried to use vitune backup in rimusic in but it didn't work have you finded solution

**Reply by fast4x** [reply]
Sorry but ViTune backup isn't compatible with RiMusic.

**Reply by Sensex00** [reply]
Hey the app ui is good and I like you are experimenting app unique ui different from vimusic. But i think you should a really find a base stable player ui like did vitune innertune ( uses m3) or even simp music ( Spotify like) .
If you find one good design. If you keen to stick to one good ui design it will good for all. 
It's just my suggestion and my wishes for finding one.

**Reply by fast4x** [reply]
I just try to keep the overall look that ViMusic has as long as possible. 
I still look for a recognizable interface that is not copied from other applications. 

Currently RiMusic allows you to use the ViMusic ui interface, similar to ViTune, or move the menu to the right, top, bottom. You can also move the miniplayer up and down, you can display the floating search icon, or use the menu with icons only. In short, with RiMusic you have the freedom to change it when you want.

thanks for appreciation.

**Comment by Abilomp** [comment]
Yes you are right .. sometimes it feel bored of the Ui.

**Reply by fast4x** [reply]
I just try to keep the overall look that ViMusic has as long as possible.
I still look for a recognizable interface that is not copied from other applications.

Currently RiMusic allows you to use the ViMusic ui interface, similar to ViTune, or move the menu to the right, top, bottom. You can also move the miniplayer up and down, you can display the floating search icon, or use the menu with icons only. In short, with RiMusic you have the freedom to change it when you want.
